x/dualstaking/keeper/migrations.go

```diff
@@ -83,81 +83,81 @@ func (m Migrator) ConvertProviderStakeToSelfDelegation(ctx sdk.Context) error {
 // 3. use the same money to delegate to both the original delegation's provider and highest staked validator
 func (m Migrator) HandleProviderDelegators(ctx sdk.Context) error {
-	delegationsInds := m.keeper.delegationFS.GetAllEntryIndices(ctx)
-	nextEpoch := m.keeper.epochstorageKeeper.GetCurrentNextEpoch(ctx)
-
-	// burn all coins from the pools
-	moduleBalance := m.keeper.bankKeeper.GetBalance(ctx, m.keeper.accountKeeper.GetModuleAddress(dualstakingtypes.BondedPoolName), m.keeper.stakingKeeper.BondDenom(ctx))
-	if !moduleBalance.IsZero() {
-		err := m.keeper.bankKeeper.BurnCoins(ctx, dualstakingtypes.BondedPoolName, sdk.NewCoins(moduleBalance))
-		if err != nil {
-			return err
-		}
-	}
-
-	if !moduleBalance.IsZero() {
-		moduleBalance = m.keeper.bankKeeper.GetBalance(ctx, m.keeper.accountKeeper.GetModuleAddress(dualstakingtypes.NotBondedPoolName), m.keeper.stakingKeeper.BondDenom(ctx))
-		err := m.keeper.bankKeeper.BurnCoins(ctx, dualstakingtypes.NotBondedPoolName, sdk.NewCoins(moduleBalance))
-		if err != nil {
-			return err
-		}
-	}
-
-	// give money back to delegators from the bonded pool
-	originalDelegations := []dualstakingtypes.Delegation{}
-	for _, ind := range delegationsInds {
-		// find the delegation and keep its original form
-		var d dualstakingtypes.Delegation
-		block, _, _, found := m.keeper.delegationFS.FindEntryDetailed(ctx, ind, nextEpoch, &d)
-		if !found {
-			continue
-		}
-
-		if d.Delegator == d.Provider || d.Provider == dualstakingtypes.EMPTY_PROVIDER {
-			continue
-		}
-
-		originalAmount := d.Amount
-
-		// zero the delegation amount in the fixation store
-		d.Amount = sdk.NewCoin(m.keeper.stakingKeeper.BondDenom(ctx), sdk.ZeroInt())
-		m.keeper.delegationFS.ModifyEntry(ctx, ind, block, &d)
-
-		// give money back from the bonded pool
-		delegatorAddr, err := sdk.AccAddressFromBech32(d.Delegator)
-		if err != nil {
-			return err
-		}
-
-		err = m.keeper.bankKeeper.MintCoins(ctx, types.ModuleName, []sdk.Coin{originalAmount})
-		if err != nil {
-			return err
-		}
-
-		err = m.keeper.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, delegatorAddr, sdk.Coins{originalAmount})
-		if err != nil {
-			return err
-		}
-
-		entry, found := m.keeper.epochstorageKeeper.GetStakeEntryByAddressCurrent(ctx, d.ChainID, d.Provider)
-		if !found {
-			continue
-		}
-		entry.DelegateTotal = entry.DelegateTotal.Sub(originalAmount)
-		m.keeper.epochstorageKeeper.ModifyStakeEntryCurrent(ctx, d.ChainID, entry)
-
-		originalDelegations = append(originalDelegations, d)
-	}
-
-	// get highest staked validator and delegate to it
-	validatorsByPower := m.keeper.stakingKeeper.GetBondedValidatorsByPower(ctx)
-	highestVal := validatorsByPower[0]
-	for _, d := range originalDelegations {
-		err := m.keeper.DelegateFull(ctx, d.Delegator, highestVal.OperatorAddress, d.Provider, d.ChainID, d.Amount)
-		if err != nil {
-			return err
-		}
-	}
+	// delegationsInds := m.keeper.delegationFS.GetAllEntryIndices(ctx)
```

Comment:
this is ugly, lets see what other chains are doing?

Reply:
done. see https://github.com/lavanet/lava/commit/9da19ae996814538aa87538ac72e3e1a4898ee88 and https://github.com/lavanet/lava/commit/f3c6bf52da12102d7e3bb16aded48f4e9636cf3b